infrastructure/terraform/components/dl/module_lambda_mesh_download.tf

```diff
@@ -0,0 +1,183 @@
+module "mesh_download" {
+  source = "https://github.com/NHSDigital/nhs-notify-shared-modules/releases/download/v2.0.24/terraform-lambda.zip"
+
+  function_name = "mesh-download"
+  description   = "A lambda function for downloading MESH messages and storing in S3"
+
+  aws_account_id = var.aws_account_id
+  component      = local.component
+  environment    = var.environment
+  project        = var.project
+  region         = var.region
+  group          = var.group
+
+  log_retention_in_days = var.log_retention_in_days
+  kms_key_arn           = module.kms.key_arn
+
+  iam_policy_document = {
+    body = data.aws_iam_policy_document.mesh_download_lambda.json
+  }
+
+  function_s3_bucket      = local.acct.s3_buckets["lambda_function_artefacts"]["id"]
+  function_code_base_path = local.aws_lambda_functions_dir_path
+  function_code_dir       = "mesh-download/target/dist"
+  function_include_common = true
+  handler_function_name   = "mesh_download.handler.handler"
+  runtime                 = "python3.9"
```

Comment:
should we use a newer runtime, like python3.13? I think 3.9 EOL was October 2025